server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GroupMappingTest.java

```diff
@@ -0,0 +1,379 @@
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP2_ADDRESS + "/" + USER_DOMAIN2)
+        .then()
+            .statusCode(201);
+
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + GROUP2_ADDRESS)
+        .then()
+            .statusCode(201);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(GROUP1_ADDRESS))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, DOMAIN1);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(USER_DOMAIN2, PASSWORD));
+
+            String processedMessage = imapMessageReader.readFirstMessageInInbox(USER_DOMAIN2, PASSWORD);
+
+            assertThat(processedMessage).contains(MESSAGE_CONTENT);
+        }
+    }
+
+    @Test
+    public void groupMappingShouldBeBiDirection() throws Exception {
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + USER_DOMAIN1)
+        .then()
+            .statusCode(201);
+
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP2_ADDRESS + "/" + USER_DOMAIN2)
+        .then()
+            .statusCode(201);
+
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + GROUP2_ADDRESS)
+        .then()
+            .statusCode(201);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(GROUP2_ADDRESS))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, DOMAIN1);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(USER_DOMAIN2, PASSWORD));
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(USER_DOMAIN1, PASSWORD));
+        }
+    }
+
+    @Test
+    public void messageShouldNotBeSentWhenGroupLoopMapping() throws Exception {
```

Comment:
messageShouldNotBeReceivedWhenGroupLoop

(The message is being sent...)

Reply:
In fact the message does not send, It may related to current bug with "503 5.5.0 No recipients specified". I will take a look